src/ragas/llms/prompt.py

```diff
@@ -109,3 +119,110 @@ def format(self, **kwargs: t.Any) -> ChatPromptTemplate:
         prompt = self.to_string()
         human_prompt = HumanMessagePromptTemplate.from_template(prompt)
         return ChatPromptTemplate.from_messages([human_prompt.format(**kwargs)])
+
+    def adapt(
+        self, language: str, llm: RagasLLM, cache_dir: t.Optional[str] = None
+    ) -> None:
+        # TODO: Add callbacks
+        cache_dir = cache_dir if cache_dir else RAGAS_CACHE_HOME
+        if os.path.exists(os.path.join(cache_dir, language, f"{self.name}.json")):
+            self._load(language, self.name, cache_dir)
+
+        prompts = []
+        for example in self.examples:
+            prompts.extend(
+                [
+                    str_translation.format(
+                        translate_to=language, input=example.get(key)
+                    )
+                    for key in self.input_keys
+                ]
+            )
+            prompts.append(
+                json_translatation.format(
+                    translate_to=language, input=example.get(self.output_key)
+                )
+                if self.output_type.lower() == "json"
+                else str_translation.format(
+                    translate_to=language, input=example.get(self.output_key)
+                )
+            )
+
+        results = [result[0].text for result in llm.generate(prompts).generations]
+        per_example_items = len(self.input_keys) + 1
+        grouped_results = [
+            results[i : i + per_example_items]
+            for i in range(0, len(results), per_example_items)
+        ]
+        assert len(grouped_results) == len(
+            self.examples
+        ), "examples and adapted examples must be of equal length"
+        for i, example in enumerate(grouped_results):
+            example_dict = {}
+            example_dict.update(
+                {k: v for k, v in zip(self.input_keys, example[: len(self.input_keys)])}
+            )
+            example_dict[self.output_key] = (
+                json_loader.safe_load(example[-1], llm)
+                if self.output_type.lower() == "json"
+                else example[-1]
+            )
+
+            self.examples[i] = example_dict
+
+        self.language = language
+
+        # TODO:Validate the prompt after adaptation
+
+    def save(self, cache_dir: t.Optional[str] = None) -> None:
+        cache_dir = cache_dir if cache_dir else RAGAS_CACHE_HOME
+        cache_dir = os.path.join(cache_dir, self.language)
+        if not os.path.exists(cache_dir):
+            os.makedirs(cache_dir)
+
+        cache_path = os.path.join(cache_dir, f"{self.name}.json")
+        with open(cache_path, "w") as file:
+            json.dump(self.to_json(), file, indent=4)
+
+    @classmethod
+    def _load(cls, language: str, name: str, cache_dir: str) -> None:
+        logging.log(logging.INFO, f"Loading {name} from {cache_dir}")
+        path = os.path.join(cache_dir, language, f"{name}.json")
+        cls(**json.load(open(path))["kwargs"])
+
```

Comment:
could you explain how this works? this is a classmethod but is not returning the cls and `adapt()` is not getting any class, so how is this working?